Sources/RTVIClientIOSGeminiLiveWebSocket/util/AudioCommon.swift

```diff
@@ -2,8 +2,38 @@ import AVFoundation
 
 // enum just for namespacing
 enum AudioCommon {
-    static var format: AVAudioFormat = {
+    static var serverAudioFormat: AVAudioFormat = {
         AVAudioFormat(commonFormat: .pcmFormatInt16, sampleRate: 24000, channels: 1, interleaved: false)!
     }()
+    
+    static let audioLevelReportingRate = 10
+    
+    static func installAudioLevelTap(onNode node: AVAudioNode, callback: @escaping (Float) -> ()) {
+        let format = node.outputFormat(forBus: 0)
+        let rate = format.sampleRate
+        let bufferSize = rate / Double(audioLevelReportingRate)
+        node.installTap(
+            onBus: 0,
+            bufferSize: AVAudioFrameCount(bufferSize),
+            format: format) { buffer, _ in
+                callback(calculateRMSAudioLevel(fromBuffer: buffer))
+            }
+    }
+
+    static func uninstallAudioLevelTap(onNode node: AVAudioNode) {
+        node.removeTap(onBus: 0)
+    }
+
+    static func calculateRMSAudioLevel(fromBuffer buffer: AVAudioPCMBuffer) -> Float {
+        let channelData = buffer.floatChannelData![0] // here we assume 1 channel and float
```

Comment:
Just out of curiosity, for `daily-js`, we have implemented the same logic as Chrome to match the audio levels we receive from WebRTC.

In that case, If multiple audio channels are used, the audio energy of a sample refers to the highest energy of any channel.

But since we are never going to use this transport and receive audio levels from WebRTC at the same time, I don't think it is necessary to use the exact same formula to calculate them.